### PR TITLE
Cherry-pick #23274 to 7.x: Synthetics docker image

### PR DESCRIPTION
**Pull request by andrewvc**
Cherry-pick of PR #23274 to 7.x branch. Original message: 

Adjusts the heartbeat docker image to install deps for heartbeat synthetics. Fixes https://github.com/elastic/beats/issues/22837

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

File: dev-tools/packaging/templates/docker/Dockerfile.tmpl [hunk 1]
```diff
@@ -8,7 +8,15 @@ FROM {{ .buildFrom }} AS home
 
 COPY beat {{ $beatHome }}
 
-RUN mkdir {{ $beatHome }}/data {{ $beatHome }}/logs && \
+{{- if (and (eq .BeatName "heartbeat") (not (contains .from "ubi-minimal"))) }}
+RUN mkdir -p {{ $beatHome }}/.node \
+    {{ $beatHome }}/.npm \
+    {{ $beatHome }}/.cache \
+    {{ $beatHome }}/.config \
+    {{ $beatHome }}/suites && \
```
Comment (jsoriano) [inline dev-tools/packaging/templates/docker/Dockerfile.tmpl:16]: There is an error here, that is making builds to fail also in master, could you please take a look?
```suggestion
    {{ $beatHome }}/suites
```

Comment (jsoriano) [reply dev-tools/packaging/templates/docker/Dockerfile.tmpl:16]: I have opened a PR to detect these issues that includes this fix: https://github.com/elastic/beats/pull/23401

**Comment by kuisathaverat** [comment]
you do not have to pet Jenkins 😄

**Comment by urso** [comment]
CI did succeed: https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats/detail/PR-23380/9/pipeline/
Status is from older run that has succeeded but marked as aborted.

**Pull request merged**